sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/GpuBatchScanExec.scala

```diff
@@ -74,26 +73,10 @@ case class GpuBatchScanExec(
         case p: KeyGroupedPartitioning =>
           if (newPartitions.exists(!_.isInstanceOf[HasPartitionKey])) {
             throw new SparkException("Data source must have preserved the original partitioning " +
-                "during runtime filtering: not all partitions implement HasPartitionKey after " +
-                "filtering")
+              "during runtime filtering: not all partitions implement HasPartitionKey after " +
+              "filtering")
           }
-
-          val newRows = new InternalRowSet(p.expressions.map(_.dataType))
-          newRows ++= newPartitions.map(_.asInstanceOf[HasPartitionKey].partitionKey())
-          val oldRows = KeyGroupedPartitioningShim.getPartitionValues(p)
-
-          if (oldRows.size != newRows.size) {
-            throw new SparkException("Data source must have preserved the original partitioning " +
-                "during runtime filtering: the number of unique partition values obtained " +
-                s"through HasPartitionKey changed: before ${oldRows.size}, after ${newRows.size}")
-          }
-
-          if (!oldRows.forall(newRows.contains)) {
-            throw new SparkException("Data source must have preserved the original partitioning " +
-                "during runtime filtering: the number of unique partition values obtained " +
-                s"through HasPartitionKey remain the same but do not exactly match")
-          }
-
+          KeyGroupedPartitioningShim.checkPartitions(p, newPartitions)
```

Comment:
Not a fan of this shim, as it sidesteps a larger issue.  `GpuBatchScanExec` was already shimmed because Spark was changing how BatchScanExec worked, and the same is true for BatchScanExec in Spark 3.3 vs. Spark 3.4. Rather than shimming this very specific item, I'd rather see a new version of GpuBatchScanExec for Spark 3.4, updated for not only the InternalRowSet change but also the other changes that have been made to GpuBatchScanExec.